recipes/pyomo/meta.yaml

```diff
@@ -0,0 +1,170 @@
+    - get_pyomo_extras=scripts.get_pyomo_extras:main
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - pyutilib >=5.3.5
+    - appdirs
+    - ply
+    - nose
+    - six >=1.4
+
+  run:
+    - python
+    - setuptools
+    - pyutilib >=5.3.5
+    - appdirs
+    - ply
+    - nose
+    - six >=1.4
+
+test:
+  imports:
+    - pyomo
+    - pyomo.bilevel
+    - pyomo.bilevel.plugins
+    - pyomo.bilevel.tests
+    - pyomo.checker
+    - pyomo.checker.plugins
+    - pyomo.checker.plugins.checkers
+    - pyomo.checker.plugins.checkers.model
+    - pyomo.checker.plugins.checkers.py3k
+    - pyomo.checker.plugins.checkers.sample
+    - pyomo.checker.tests
+    - pyomo.core
+    - pyomo.core.base
+    - pyomo.core.beta
+    - pyomo.core.data
+    - pyomo.core.plugins
+    - pyomo.core.plugins.data
+    - pyomo.core.plugins.transform
+    - pyomo.core.preprocess
+    - pyomo.core.tests
+    - pyomo.core.tests.data
+    - pyomo.core.tests.diet
+    - pyomo.core.tests.examples
+    - pyomo.core.tests.transform
+    - pyomo.core.tests.unit
+    - pyomo.dae
+    - pyomo.dae.plugins
+    - pyomo.dae.tests
+    - pyomo.data
+    - pyomo.environ
+    - pyomo.environ.tests
+    - pyomo.gdp
+    - pyomo.gdp.plugins
+    - pyomo.gdp.tests
+    - pyomo.mpec
+    - pyomo.mpec.plugins
+    - pyomo.mpec.tests
+    - pyomo.neos
+    - pyomo.neos.plugins
+    - pyomo.openopt
+    - pyomo.openopt.plugins
+    - pyomo.openopt.tests
+    - pyomo.opt
+    - pyomo.opt.base
+    - pyomo.opt.blackbox
+    - pyomo.opt.parallel
+    - pyomo.opt.plugins
+    - pyomo.opt.problem
+    - pyomo.opt.results
+    - pyomo.opt.solver
+    - pyomo.opt.testing
+    - pyomo.opt.tests
+    - pyomo.opt.tests.base
+    - pyomo.opt.tests.blackbox
+    - pyomo.opt.tests.solver
+    - pyomo.pysp
+    - pyomo.pysp.plugins
+    - pyomo.pysp.scenariotree
+    - pyomo.pysp.smps
+    - pyomo.pysp.tests
+    - pyomo.pysp.tests.examples
+    - pyomo.pysp.tests.scenariotreemanager
+    - pyomo.pysp.tests.smps
+    - pyomo.pysp.tests.unit
+    - pyomo.pysp.util
+    - pyomo.repn
+    - pyomo.repn.beta
+    - pyomo.repn.plugins
+    - pyomo.repn.plugins.ampl
+    - pyomo.repn.tests
+    - pyomo.repn.tests.ampl
+    - pyomo.scripting
+    - pyomo.scripting.plugins
+    - pyomo.solvers
+    - pyomo.solvers.plugins
+    - pyomo.solvers.plugins.converter
+    - pyomo.solvers.plugins.smanager
+    - pyomo.solvers.plugins.solvers
+    - pyomo.solvers.plugins.testdriver
+    - pyomo.solvers.tests
+    - pyomo.solvers.tests.io
+    - pyomo.solvers.tests.io.fd
+    - pyomo.solvers.tests.mip
+    - pyomo.solvers.tests.pickling
+    - pyomo.util
+    - pyomo.util.tests
+    - pyomo.version
+    - pyomo.version.tests
+
+  commands:
+    - runbenders --help
+    - evaluate_xhat --help
+    - runph --help
+    - runef --help
+    - pysp2smps --help
+    - phsolverserver --help
+    - scenariotreeserver --help
+    - computeconf --help
+    - test.pyomo --help
+    - pyomo --help
```

Comment:
I'm pretty comfortable with this set of tests.

A lot of these entry points require the use of Pyro, and many of the
scripts generate errors when Pyro is not installed. However, I do not want
to treat Pyro as a required dependency for Pyomo.  In practice, it is not
required.

I've highlighted this behavior for the Pyomo developer team, and it'll be
fixed during the next major release.  But IMO this isn't a reason to hold
up the release.

Pyomo is extensively tested, though the tests aren't publicly available.
The project is moving to github shortly, so there will be publicly
available tests in future releases as well.

On Tue, Apr 26, 2016 at 9:15 PM, jakirkham <notifications@github.com> wrote:

> In recipes/pyomo/meta.yaml
> <https://github.com/conda-forge/staged-recipes/pull/449#discussion_r61190815>
> :
>
> > +    - pyomo.util
> > +    - pyomo.util.tests
> > +    - pyomo.version
> > +    - pyomo.version.tests
> > +
> > +  commands:
> > +    - runbenders --help
> > +    - evaluate_xhat --help
> > +    - runph --help
> > +    - runef --help
> > +    - pysp2smps --help
> > +    - phsolverserver --help
> > +    - scenariotreeserver --help
> > +    - computeconf --help
> > +    - test.pyomo --help
> > +    - pyomo --help
>
> Maybe we should test more of the entrypoints above. Am a little worried
> that as there are so many it would be easy to miss something in there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/conda-forge/staged-recipes/pull/449/files/62899347856be9cfe0db02879d5935a43c0455cd#r61190815>
>